proto/structured.proto

```diff
@@ -104,3 +105,224 @@ message CreateTableResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// A ConditionalPutTableRowRequest.
+//
+// - Returns true and sets row if exp_row equals existing row.
+// - If key doesn't exist and exp_row is empty, sets row.
+// - If key exists and exp_row is empty, sets row.
+// - Otherwise, returns error and the actual value of the row in the response.
+message ConditionalPutTableRowRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional TableKey key = 2 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // The values to put.
+  optional TableRow row = 3 [(gogoproto.nullable) = false];
+  optional TableRow exp_row = 4[(gogoproto.nullable) = false];
+}
+
+// A ConditionalPutTableRowResponse is the return value from the
+// ConditionalPut request.
+message ConditionalPutTableRowResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional TableRow row = 2[(gogoproto.nullable) = false];
+}
+
+// A DeleteTableRowRequest deletes the entire row.
+message DeleteTableRowRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional TableKey key = 2 [(gogoproto.nullable) = false];
+}
+
+// A DeleteTableRowResponse.
+message DeleteTableRowResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// A DeleteTableRowRangeRequest deletes the specified rows.
+message DeleteTableRowRangeRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Delete *all* entries between Key (inclusive) and EndKey
+  // (exclusive). Must be > 0
+  optional string index = 2 [(gogoproto.nullable) = false];
+  optional TableKey key = 3 [(gogoproto.nullable) = false];
+  optional TableKey end_key = 4 [(gogoproto.nullable) = false];
+  optional int64 max_entries_to_delete = 5 [(gogoproto.nullable) = false];
+}
+
+// A DeleteTableRowRangeResponse is the response.
+message DeleteTableRowRangeResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Number of entries deleted.
+  optional int64 num_deleted = 2 [(gogoproto.nullable) = false];
+}
+
+// A ScanTableRequest. It specifies the start and end keys for the scan
+// and the maximum number of results.
+message ScanTableRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Must be > 0.
+  optional string index = 2 [(gogoproto.nullable) = false];
+  optional TableKey key = 3 [(gogoproto.nullable) = false];
+  optional TableKey end_key = 4 [(gogoproto.nullable) = false];
+  optional int64 max_results = 5 [(gogoproto.nullable) = false];
+  // get specified columns from the rows.
+  repeated string column_names = 6;
+}
+
+// A ScanTableResponse is the response.
```

Comment:
The comments need another pass. This comment only makes sense in context with the `ScanTableRequest` comment which is a dozen lines above.